### DISCUSSION TITLE
Netboot ChromeOS Flex

### DISCUSSION THREAD

**Discussion by bshogeman** [ipxe/ipxe#1343] [Q&A]
I know it is possible to netboot ChromeOS Flex And it seems like iventoy is just doing it. (not tested it myself only seen screenshots)
But then if they can do it maybe it is also possible with iPXE.

It is just a raw image with grub and syslinux config files inside.

menuentry "local image A" {
  linux /syslinux/vmlinuz.A quiet init=/sbin/init rootwait ro noresume loglevel=7 noinitrd console= kvm-intel.vmentry_l1d_flush=always  i915.modeset=1 cros_efi       root=PARTUUID=AB9474B5-FEE2-CD46-BE94-E9F2402A54FE
}

# FIXME: usb doesn't support verified boot for now
menuentry "Alternate USB Boot" {
  linux (hd0,3)/boot/vmlinuz quiet init=/sbin/init rootwait ro noresume loglevel=7 noinitrd console= kvm-intel.vmentry_l1d_flush=always  root=PARTUUID=A6608836-844F-6044-A9FF-D32F90F349D5 i915.modeset=1 cros_efi
}

OR

label chromeos-usb.A
  menu label chromeos-usb.A
  kernel vmlinuz.A
  append quiet init=/sbin/init rootwait ro noresume loglevel=7 noinitrd console= kvm-intel.vmentry_l1d_flush=always  root=PARTUUID=A6608836-844F-6044-A9FF-D32F90F349D5 i915.modeset=1 cros_legacy

label chromeos-vusb.A
  menu label chromeos-vusb.A
  kernel vmlinuz.A
  append quiet init=/sbin/init rootwait ro noresume loglevel=7 noinitrd console= kvm-intel.vmentry_l1d_flush=always  dm_verity.error_behavior=3 dm_verity.max_bios=-1 dm_verity.dev_wait=1 root=/dev/dm-0       i915.modeset=1 cros_legacy dm="1 vroot none ro 1,0 4710400 verity payload=PARTUUID=A6608836-844F-6044-A9FF-D32F90F349D5 hashtree=PARTUUID=A6608836-844F-6044-A9FF-D32F90F349D5 hashstart=4710400 alg=sha256 root_hexdigest=58355605aab211507da536c799e0874118c2a9db08d835d75f40aa99186b16c9 salt=2860d71c5c7ff236b9700361f9f1c9c0fcd929e300ee7ab59b0051feec6ee117"

label chromeos-hd.A
  menu label chromeos-hd.A
  kernel vmlinuz.A
  append quiet init=/sbin/init rootwait ro noresume loglevel=7 noinitrd console= kvm-intel.vmentry_l1d_flush=always  root=PARTUUID=AB9474B5-FEE2-CD46-BE94-E9F2402A54FE i915.modeset=1 cros_legacy

label chromeos-vhd.A
  menu label chromeos-vhd.A
  kernel vmlinuz.A
  append quiet init=/sbin/init rootwait ro noresume loglevel=7 noinitrd console= kvm-intel.vmentry_l1d_flush=always  dm_verity.error_behavior=3 dm_verity.max_bios=-1 dm_verity.dev_wait=1 root=/dev/dm-0       i915.modeset=1 cros_legacy dm="1 vroot none ro 1,0 4710400 verity payload=PARTUUID=AB9474B5-FEE2-CD46-BE94-E9F2402A54FE hashtree=PARTUUID=AB9474B5-FEE2-CD46-BE94-E9F2402A54FE hashstart=4710400 alg=sha256 root_hexdigest=58355605aab211507da536c799e0874118c2a9db08d835d75f40aa99186b16c9 salt=2860d71c5c7ff236b9700361f9f1c9c0fcd929e300ee7ab59b0051feec6ee117"

--
Getting the img file to test with? https://dl.google.com/chromeos-flex/images/latest-massdeploy.bin.zip
Or choose one from: https://dl.google.com/dl/edgedl/chromeos/recovery/cloudready_recovery2.json
These images are not the same.

Somone who knows how?

**Comment by NiKiZe** [comment]
Just had an encounter with shredos today which looks similar, simply chain to the kernel with the suggested cmdline parameters?
